### DISCUSSION TITLE
SnowflakeSqlApiHook should mask Private Key text

### DISCUSSION THREAD

**Discussion by erdos2n** [apache/airflow#36625] [General]
### Description

When adding the private key text the value gets added to the extras field and is exposed. Can we get this masked or not added to the extras?

https://github.com/apache/airflow/blob/c47dcc59e5a0766d6b390bc8ceafc8ab6b762c88/airflow/providers/snowflake/hooks/snowflake.py#L116

### Use case/motivation

I would like the private key to stay masked on the Airflow UI

### Related issues

Security Vulnerability for users

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
Please avoid creating an issue where you are asking questions especially this way.

And if you **think** (after consulting our Security Model that is) that something is a security vulnerability, then you should NEVER create a public issue but you should follow our security policy https://github.com/apache/airflow/security/policy and report it responsibly. Opening a public issue and stating in it "This is a security vulnerability" is generally NEVER a good idea. 

Pleaso don't do it it in the future.

But in this case - no. It's not a security vullnerability and you will find answer to it in our security model. I generally recommend you to follow it before you claim otherwise (and then behave responsibly by reporting a secuirty issue in a responsible way). 

The reason why it's not a security issue is that Connection Editing User is highly priviledged - see our security model https://airflow.apache.org/docs/apache-airflow/stable/security/security_model.html . Such user has access to all connections and all authentication for those. The fact that some fields (like passwords) in the UI are masked is merely to pretend a by-standard to look over the shoulder of that person who has priviledges and can look up the values anyway if they want (if you look at inspect page of the connection page, the passwords configured for that connection are still visible if you change property of those fields using Inspect  - they are merely masked by setting "password" property. And this is by design. 

So no - this is not a security feature but if you think that this field is worthy to be masked, you are free to submit a PR for that - I think it's not worthy to be honest but if you want - feel free. and there is no need to open issues.
